topdown/cache/cache.go

```diff
@@ -165,3 +165,89 @@ func (c *cache) maxSizeBytes() int64 {
 	}
 	return *c.config.InterQueryBuiltinCache.MaxSizeBytes
 }
+
+// Note(philipc): The non-deterministic builtins cache exists to allow
+// decision logging to preserve the results of different non-deterministic
+// builtin function calls. This paves the way for later reconstruction and
+// replay of policies from their decision logs.
+// There are constraints however, namely that we only save the first
+// result we see for a given set of input parameters. This means that for
+// builtins like http.send, we assume that the outside world is able to
+// return arbitrary content, but that the result will be idempotent across
+// successive calls of that builtin with the same inputs. Otherwise, we
+// begin to descend into selective tracing of just the non-deterministic
+// builtins.
```

Comment:
Let's update the comment at some point regarding multiple calls there (see my comment on the issue).